### PR TITLE
5343 upgrade django in mitxonline

### PR DESCRIPTION
**Pull request by cp-at-mit**
### What are the relevant tickets?
https://github.com/mitodl/hq/issues/5343

### Description (What does it do?)
Updates Django in MITx Online along with libraries that must be updated in order to support the new version of Django.

### Additional Context
Changes from using django-fsm to using viewflow for Order state management.

Updated libraries can be seen in the pyproject.toml file in this PR.

### CODE REVIEW

File: wget-log
Comment (jkachel) [inline wget-log:1]: What is this?

**Review by jkachel — CHANGES_REQUESTED** [review]
Some minor things to take care of - looks like there's a stray file that got added to the repo, and can the `version` be pulled from the Compose files? `version` is harmless but also deprecated so Docker emits a message about it. 

Otherwise: I tested a lot of the system and a lot of it looks good to me. Automated tests passed and I was able to sign in, add courses and programs, set up a product, set up discounts, set up financial assistance, test purchasing and refunding, and test the staff dashboard. There were some things that may need addressing:
- In the financial assistance form, the currency selector didn't render with any options. This *appears* to be just pulling from the CurrencyExchangeRate table - which is populated on my instance - but maybe there's something else that needs to be fixed here. It does work on RC so it may also just be my instance.
- I had some issues with the staff dashboard (`ResizeObserver loop completed with undelivered notifications`) but it does appear to work? Your changes don't touch the staff dashboard so this is probably just me, but am noting it.

Some things I did not test:
- I did not test the Google Sheets integration because I don't have that set up to test with at the moment.
- I also didn't test the edX integration because I don't have a reasonable Tutor instance running. It did appear to try to hit it, though, since it failed a whole bunch trying to do that.
- Furthermore, I didn't test HubSpot because I also don't have this set up (and actually don't know how to do this).

**Comment by cp-at-mit** [comment]
> * ResizeObserver loop completed with undelivered notifications

Yeah I am getting that ResizeObserver error as well, but it doesn't really point to anything changed

**Comment by cp-at-mit** [comment]
> In the financial assistance form, the currency selector didn't render with any options. This appears to be just pulling from the CurrencyExchangeRate table - which is populated on my instance - but maybe there's something else that needs to be fixed here. It does work on RC so it may also just be my instance.

That appears to be caused by either: not having the fields for the flexible pricing form defined in the CMS, or they are defined but out of order.

**Comment by jkachel** [comment]
> > * ResizeObserver loop completed with undelivered notifications
> 
> Yeah I am getting that ResizeObserver error as well, but it doesn't really point to anything changed. Looks like there is a CORS error in the browser console that might be associated. But also, both of those pages are controlled by the CMS, so maybe I need to update Wagtail.

I don't see any CORS errors on my end.

After doing some more debugging, this might (_might_) be a react-router-dom v6 issue? I found some stuff about this [in this Stack Overflow post](https://stackoverflow.com/questions/76187282/react-resizeobserver-loop-completed-with-undelivered-notifications). But this isn't really conclusive - maybe Refine does something with this behind the scenes. We're pretty out of date on even the 3.x branch of Refine now, though, so maybe there's something to that. (We're 3.22.1 and latest 3.x is 3.103.0 - and that's deprecated in favor of 4.x.)

I don't see the error on RC or prod and this really seems to be out of scope for this PR (especially after testing some more - this really seems to be an issue with the Refine stack here, not with any changes in the Django app) so it might be best to punt on this, especially given that it'd also be a good idea to update to Refine 4.

**Review by annagav — COMMENTED** [review]
I don't seem to be able to go to http://mitxonline.odl.local:8013/cms/

**Comment by cp-at-mit** [comment]
> I don't seem to be able to go to http://mitxonline.odl.local:8013/cms/

Are you receiving an error?

**Comment by cp-at-mit** [comment]
@annagav can you try running a `docker compose build`?

**Comment by annagav** [comment]
> @annagav can you try running a `docker compose build`?

I did. But I can try again.

**Comment by cp-at-mit** [comment]
@jkachel are you able to load the CMS while using this branch?

**Comment by jkachel** [comment]
> @jkachel are you able to load the CMS while using this branch?

Yes - there were new migrations, though, that needed to be applied.

**Comment by cp-at-mit** [comment]
@jkachel so you just needed to run `docker-compose run web ./manage.py migrate` ?

**Comment by jkachel** [comment]
> @jkachel so you just needed to run `docker-compose run web ./manage.py migrate` ?

Yes, that got it working for me. I did have to do it manually though.

**Review by annagav — APPROVED** [review]
Seems to works as expected 👍 User dashboard, cms, admin.
Google sheets request work too.

**Comment by jkachel** [comment]
> > In the financial assistance form, the currency selector didn't render with any options. This appears to be just pulling from the CurrencyExchangeRate table - which is populated on my instance - but maybe there's something else that needs to be fixed here. It does work on RC so it may also just be my instance.
> 
> That appears to be caused by either: not having the fields for the flexible pricing form defined in the CMS, or they are defined but out of order.

I figured this out finally - my local version needed the `fixtree` command to be run because I was getting an error when creating a financial assistance form (`AttributeError: 'NoneType' object has no attribute '_inc_path'`). I was able to create a form, then add the fields manually and submit a new request. 

However, aren't the form fields supposed to be added automatically? Did this stop at some point? FWIW, RC doesn't do it either so this is more for my curiosity.

**Review by jkachel — APPROVED** [review]
Looks OK once I figured out some issues with Wagtail on my end. I do have questions about form fields in financial assistance forms but since this isn't a changed behavior I'm not concerned about addressing it in this PR.

**Pull request merged**